File: web/src/test/java/hu/bsstudio/team/UpdateTeamHandlerTest.java
package hu.bsstudio.team;

import static org.mockito.Mockito.when;
import static org.mockito.MockitoAnnotations.initMocks;

import hu.bsstudio.team.model.DetailedTeam;
import hu.bsstudio.team.model.Team;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mockito.Mock;
import org.springframework.test.web.reactive.server.WebTestClient;
import org.springframework.web.reactive.function.server.RouterFunctions;
import reactor.core.publisher.Mono;

final class UpdateTeamHandlerTest {

    @Mock
    private TeamService mockService;

    private WebTestClient webTestClient;

    @BeforeEach
    void setUp() {
        initMocks(this);
        final var underTest = new UpdateTeamHandler(mockService);
        final var routerFunction = RouterFunctions.route()
            .POST("/test", underTest).build();
        this.webTestClient = WebTestClient.bindToRouterFunction(routerFunction).build();
    }

    @Test
    void shouldReturnDetailedTeamWithOkStatus() {
        final Team team = Team.builder().build();
        final var detailedTeam = DetailedTeam.builder().build();
        when(mockService.updateTeam(team))
            .thenReturn(Mono.just(detailedTeam));

        webTestClient.post().uri("/test").bodyValue(team).exchange()
            .expectStatus().isOk()
            .expectBody(DetailedTeam.class).isEqualTo(detailedTeam);
    }
}

File: web/src/test/kotlin/hu/bsstudio/robonaut/team/UpdateTeamHandlerTest.kt
package hu.bsstudio.robonaut.team

import hu.bsstudio.robonaut.team.model.DetailedTeam
import hu.bsstudio.robonaut.team.model.Team
import io.mockk.MockKAnnotations
import io.mockk.every
import io.mockk.impl.annotations.MockK
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.springframework.test.web.reactive.server.WebTestClient
import org.springframework.test.web.reactive.server.expectBody
import org.springframework.web.reactive.function.server.RouterFunctions
import reactor.core.publisher.Mono

internal class UpdateTeamHandlerTest {
    
    @MockK
    private lateinit var mockService: TeamService
    private lateinit var webTestClient: WebTestClient
    
    @BeforeEach
    fun setUp() {
        MockKAnnotations.init(this)
        val underTest = UpdateTeamHandler(mockService)
        val routerFunction = RouterFunctions.route()
            .POST("/test", underTest).build()
        webTestClient = WebTestClient.bindToRouterFunction(routerFunction).build()
    }

    @Test
    fun `should return DetailedTeam with OK status`() {
        val team = Team.builder().build()
        val detailedTeam = DetailedTeam.builder().build()
        every { mockService.updateTeam(team) } returns Mono.just(detailedTeam)

        webTestClient.post().uri("/test").bodyValue(team).exchange()
            .expectStatus().isOk
            .expectBody<DetailedTeam>().isEqualTo(detailedTeam)
    }
}
